File: src/LeetCode/ListNode.java
package LeetCode;

public class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }

    public static ListNode makeList(int[] nums) {
        ListNode head = new ListNode(nums[0]);
        ListNode curr = head;
        for (int i = 1; i < nums.length; i++) {
            curr.next = new ListNode(nums[i]);
            curr = curr.next;
        }
        return head;
    }

    public static void showList(ListNode head) {
        while (head != null) {
            System.out.print(head.val + " ");
            head = head.next;
        }
    }
}

File: src/LeetCode-in-Go/utils/ListNode.go
package main

import "fmt"

type ListNode struct {
	Val int
	Next *ListNode
}

func MakeList(nums []int) *ListNode {
	if len(nums) == 0 {
		return nil
	}
	dummy := &ListNode{-1, nil}
	curr := dummy
	for _, val := range nums {
		curr.Next = &ListNode{val, nil}
		curr = curr.Next
	}
	return dummy.Next
}

func PrintList(head *ListNode) {
	for head != nil {
		fmt.Printf("%d ", head.Val)
		head = head.Next
	}
}

func main() {
	var nums []int = []int{1,2,3,4,5}
	head := MakeList(nums)
	PrintList(head)
}


